### PR TITLE
Update go to 1.17 for policy-framework repo

### PR DESCRIPTION
**Pull request by ycao56**
Signed-off-by: Yu Cao <ycao@redhat.com>

### CODE REVIEW

File: ci-operator/config/stolostron/governance-policy-framework/stolostron-governance-policy-framework-release-2.5.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ build_root:
   image_stream_tag:
     name: builder
     namespace: stolostron
-    tag: go1.16-linux
+    tag: go1.17-linux
```
Comment (dhaiducek) [inline ci-operator/config/stolostron/governance-policy-framework/stolostron-governance-policy-framework-release-2.5.yaml:10]: The `Dockerfile.e2etest` instance below needs to be updated also.

**Review by dhaiducek — CHANGES_REQUESTED** [review]

File: ci-operator/config/stolostron/governance-policy-framework/stolostron-governance-policy-framework-release-2.5.yaml (hunk 1 above)
Comment (ycao56) [reply ci-operator/config/stolostron/governance-policy-framework/stolostron-governance-policy-framework-release-2.5.yaml:10]: Yeah, the `main` branch has been updated. `release-2.5` will be updated after being fast forwarded

Comment (ycao56) [reply ci-operator/config/stolostron/governance-policy-framework/stolostron-governance-policy-framework-release-2.5.yaml:10]: done

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**